test/testpaint/compat.c

```diff
@@ -129,6 +129,10 @@ int map_element_get_direction(const rct_map_element *element) {
     return element->type & MAP_ELEMENT_DIRECTION_MASK;
 }
 
+int map_element_get_direction_with_offset(const rct_map_element *element, int offset) {
+    return ((element->type & MAP_ELEMENT_DIRECTION_MASK) + offset) % 4;
```

Comment:
I opted for making the `offset` uint8 instead, since it failed to compile otherwise.